demo/generate_demo_repo.sh

```diff
@@ -12,7 +12,11 @@ set -e
 ############
 readonly VERSION=1.0.0
 readonly SCRIPT_NAME=${0##*/}
-readonly SCRIPT_DIR=$(dirname $(realpath -e "$0"))
+if [[ "$OSTYPE" =~ ^darwin ]]; then
+    readonly SCRIPT_DIR=$(dirname $(realpath "$0"))
+else
+    readonly SCRIPT_DIR=$(dirname $(realpath -e "$0"))
```

Comment:
IMO, we can simply drop this variant. `-e` requires that all components of the path exist. Normal behavior is to enforce everything /except/ the last component.

In this case, we /know/ that all of the components exist (it's coming from `$0` after all).